HORUS/SOURCES/SRC/WIDGETS/Yaapu/main.lua

```diff
@@ -1001,8 +1012,178 @@ local function resetHash()
+	-- telemetry.hSpeed
+	local hSpeed = mavsdk.getVfrGroundSpeed()
+	if hSpeed ~= nil then telemetry.hSpeed =  hSpeed * 10 end -- from m/s to dm/s
+	-- telemetry.yaw
+	local yaw = mavsdk.getVfrHeadingDeg()
+	if yaw ~= nil then telemetry.yaw = yaw end
+	-- telemetry.flightMode
+	local fm = mavsdk.getFlightMode()
+	if fm ~= nil then telemetry.flightMode = fm + 1 end -- needs to be incremented by one from MAVLink standard output
+	-- telemetry.simpleMode
+	telemetry.simpleMode = 0
+	-- telemetry.statusArmed and telemetry.landComplete
+	local armed = mavsdk.isArmed()
+	if armed ~= nil then
+		if armed then
+		  telemetry.statusArmed = 1
+		  telemetry.landComplete = 1
+		else
+		  telemetry.statusArmed = 0
+		  telemetry.landComplete = 0
+		end
+        if not status.homeGood then
+		  MarkHome()
+		end
+	end
+	-- telemetry.ekfFailsafe
+	telemetry.ekfFailsafe = 0
+	-- telemetry.imuTemp -- [°C] value not yet parsed by OlliW
+	-- telemetry.numSats
+	local gpssat = mavsdk.getGpsSat()
+	if gpssat ~= nil then
+	  if gpssat > 99 then gpssat = 0 end
+	  telemetry.numSats = gpssat
+	end
+	-- telemetry.gpsStatus
+	local gpsStatus = mavsdk.getGpsFix()
+	if gpsStatus ~= nil then
+		telemetry.gpsStatus = gpsStatus
+	    if not status.homeGood then
+		  MarkHome()
+		end
+	end
+	-- telemetry.gpsHdopC
+	local Hdop = mavsdk.getGpsHDop() -- here NOT divided with 10 to keep the original dilution scale
+	if Hdop ~= nil then telemetry.gpsHdopC = Hdop end
+	-- telemetry.gpsAlt
+	local gpsAlt = mavsdk.getGpsAltitudeMsl()
+	if gpsAlt ~= nil then telemetry.gpsAlt = gpsAlt * 10 end -- from m to dm
+	-- telemetry.batt1volt
+	local batt1Volt = mavsdk.getBatVoltage()
+	if batt1Volt ~= nil then telemetry.batt1volt = batt1Volt * 10 end -- from V to dV
+	-- telemetry.batt1current
+	local BatCurrent = mavsdk.getBatCurrent()
+	if BatCurrent ~= nil then telemetry.batt1current = BatCurrent end
+	-- telemetry.batt1mah
+	local BatCharge = mavsdk.getBatChargeConsumed()
+	if BatCharge ~= nil then telemetry.batt1mah = BatCharge end
+	-- telemetry.batt2volt
+	local batt2Volt = mavsdk.getBat2Voltage()
+	if batt2Volt ~= nil then telemetry.batt2volt = batt2Volt * 10 end -- from V to dV
+	-- telemetry.batt2current
+	local Bat2current = mavsdk.getBat2Current()
+	if Bat2current ~= nil then telemetry.batt2current = Bat2current end
+	-- telemetry.batt2mah
+	local Bat2Charge = mavsdk.getBat2ChargeConsumed()
+	if Bat2Charge ~= nil then telemetry.batt2mah = Bat2Charge end
+	-- telemetry.lat and telemetry.lon
+	local latlon = mavsdk.getGpsLatLonInt()
+	if latlon.lat ~= nil then
+	  telemetry.lat = latlon.lat / 10000000	-- converted to degrees.fraction
+	end
+	if latlon.lon ~= nil then
+	  telemetry.lon = latlon.lon / 10000000 -- converted to degrees.fraction
+	end
+	-- telemetry.homeAngle and telemetry.homeDist
+	if status.homeGood and status.homelon ~= nil and status.homelat ~= nil and telemetry.lon ~= nil and telemetry.lat ~= nil then
```

Comment:
this is an expensive calculation should be done at a slower rate, refactor it in a dedicated function and move it to background